apps/emqx_resource/src/schema/emqx_resource_schema.erl

```diff
@@ -149,4 +151,17 @@ max_queue_bytes(default) -> ?DEFAULT_QUEUE_SIZE_RAW;
 max_queue_bytes(required) -> false;
 max_queue_bytes(_) -> undefined.
 
+queue_mode(type) -> enum([memory_only, volatile_offload]);
```

Comment:
rename 'queue' to buffer everywhere?
the max_queue_bytes should be named max_buffer_bytes, and move 'max_queue_bytes' to an aliase.

Reply:
:heavy_check_mark: